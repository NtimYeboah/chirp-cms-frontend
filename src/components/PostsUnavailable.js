/**
 * Created by ntimobedyeboah on 4/16/17.
 */
import React, { Component } from 'react';

class PostsUnavailable extends Component {
    render() {
        return (
            <section className="section">
                <div className="container">
                    <div className="columns">
                        <div className="column is-12">
                            <div className="card">
                                <header className="card-header">
                                    <p className="card-header-title">
                                        No Post Available
                                    </p>
                                    <a className="card-header-icon">
                                        <span className="icon">
                                            <i className="fa fa-angle-down"></i>
                                        </span>
                                    </a>
                                </header>
                                <div className="card-content">
                                    <div className="content">
                                        If you are the admin, add posts.
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}

export default PostsUnavailable;