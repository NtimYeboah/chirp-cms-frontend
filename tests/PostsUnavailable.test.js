/**
 * Created by ntimobedyeboah on 4/19/17.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import PostsUnavailable from './../src/components/PostsUnavailable';

describe('PostsUnavailabe Snapshot', () => {
    test('it render PostsUnavailable component and take snapshot', () => {
        const component = renderer.create(<PostsUnavailable />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});


describe('PostUnavailable rendered output', () => {
    test('it should display message to user', () => {
        const component = shallow(<PostsUnavailable />);

        expect(component.find('p.card-header-title').text()).toEqual('No Post Available');
        expect(component.find('div.content').text()).toEqual('If you are the admin, add posts.');
    });
});

