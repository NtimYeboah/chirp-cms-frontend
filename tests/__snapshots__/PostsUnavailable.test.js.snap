exports[`PostsUnavailabe Snapshot it render PostsUnavailable component and take snapshot 1`] = `
<section
  className="section">
  <div
    className="container">
    <div
      className="columns">
      <div
        className="column is-12">
        <div
          className="card">
          <header
            className="card-header">
            <p
              className="card-header-title">
              No Post Available
            </p>
            <a
              className="card-header-icon">
              <span
                className="icon">
                <i
                  className="fa fa-angle-down" />
              </span>
            </a>
          </header>
          <div
            className="card-content">
            <div
              className="content">
              If you are the admin, add posts.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
